File: app/src/main/java/com/kidozh/discuzhub/utilities/CharsetUtils.java
package com.kidozh.discuzhub.utilities;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;

public class CharsetUtils {
    public static String EncodeStringByCharset(String str,String charset){
        try{
            return URLEncoder.encode(str,charset);
        }
        catch (UnsupportedEncodingException e){
            e.printStackTrace();
            return str;
        }
    }
}

File: app/src/main/java/com/kidozh/discuzhub/utilities/CharsetUtils.kt
package com.kidozh.discuzhub.utilities

import java.io.UnsupportedEncodingException
import java.net.URLEncoder

object CharsetUtils {
    @JvmStatic
    fun EncodeStringByCharset(str: String, charset: String?): String {
        return try {
            URLEncoder.encode(str, charset)
        } catch (e: UnsupportedEncodingException) {
            e.printStackTrace()
            str
        }
    }
}